docs/topics/data-analysis/data-analysis-connect-to-db.md

```diff
@@ -1,22 +1,167 @@
+schema information for each table in the database:
+
+```kotlin
+val dataschemas = DataFrame.getSchemaForAllSqlTables(dbConfig)
+
+dataschemas.forEach { 
+    println("---Yet another table schema---")
+    println(it)
+    println()
+}
+```
+
+For more information on connecting to SQL databases, see [Read from SQL databases in the DataFrame documentation](https://kotlin.github.io/dataframe/readsqldatabases.html).
+
+## Retrieve and manipulate data
+
+After [establishing a connection to an SQL database](#connect-to-database), you can retrieve and manipulate data in Kotlin Notebook, utilizing the DataFrame library. 
+You can use `readSqlTable` to retrieve data. To manipulate data, you can use methods, such as [`.filter`](https://kotlin.github.io/dataframe/filter.html), [`.groupBy`](https://kotlin.github.io/dataframe/groupby.html), 
+and [`.convert`](https://kotlin.github.io/dataframe/convert.html). 
+
+Let's look at an example of connecting to an IMDB database and retrieving data about movies directed by Quentin Tarantino:
+
+1. **Fetch data from a table:** Use the `readSqlTable` function to retrieve data from the "movies" table, setting `limit` 
+to restrict the query to the first 100 records for efficiency:
+
+```kotlin
+val dfs = DataFrame.readSqlTable(dbConfig, tableName = "movies", limit = 100)
+```
+
+2. **Execute SQL query:**
+Use an SQL query to retrieve a specific dataset related to movies directed by Quentin Tarantino. 
+This query selects movie details and combines genres for each movie:
+
+```kotlin
+val props = Properties()
+props.setProperty("user", USER_NAME)
+props.setProperty("password", PASSWORD)
+
+val TARANTINO_FILMS_SQL_QUERY = """
+    SELECT name, year, rank, GROUP_CONCAT(genre) as "genres"
+    FROM movies JOIN movies_directors ON movie_id = movies.id
+    JOIN directors ON directors.id=director_id LEFT JOIN movies_genres ON movies.id = movies_genres.movie_id
+    WHERE directors.first_name = "Quentin" AND directors.last_name = "Tarantino"
+    GROUP BY name, year, rank
+    ORDER BY year
+    """
+
+var dfTarantinoMovies: DataFrame<*>
+
+DriverManager.getConnection(URL, props).use { connection ->
+   connection.createStatement().use { st ->
+      st.executeQuery(TARANTINO_FILMS_SQL_QUERY).use { rs ->
+         val dfTarantinoFilmsSchema = DataFrame.getSchemaForResultSet(rs, connection)
+         dfTarantinoFilmsSchema.print()
+
+         dfTarantinoMovies = DataFrame.readResultSet(rs, connection)
+         dfTarantinoMovies
+      }
+   }
+}
+```
+
+3. **Manipulate and filter data:**
+After fetching the Tarantino movies dataset, you can further manipulate and filter the data.
+
+```kotlin
+val df = dfTarantinoMovies
+    .fillNA { year }.with { 0 }
+    .convert { year }.toInt()
+    .filter { year > 2000 }
+```
+
+The resulting output is a DataFrame where missing values in the year column are replaced with 0 using the 
```

Comment:
```suggestion
The resulting output is a data frame where missing values in the year column are replaced with 0 using the 
```

Reply:
Here we are referring to a data frame as a simple noun, different from DataFrame, the name of the library. I see this issue across the tutorials, please change it.

Reply:
as discussed - I'll keep it as DataFrame as that's the name of the instance to avoid confusion